repl/tourist.py

```diff
@@ -0,0 +1,90 @@
+import json
+from subprocess import Popen, PIPE
+
+
+class Tourist:
+    def __init__(self):
+        self.proc = Popen(["target/debug/tourist", "serve"],
+                          stdout=PIPE, stdin=PIPE)
+
+    def __rpc(self, method, args):
+        msg = {
+            "jsonrpc": "2.0",
+            "method": method,
+            "params": args,
+            "id": 1
+        }
+        self.proc.stdin.write((json.dumps(msg) + "\n").encode("utf-8"))
+        self.proc.stdin.flush()
+        return json.loads(self.proc.stdout.readline().decode("utf-8"))
+
+    def list_tours(self):
+        return self.__rpc("list_tours", [])
+
+    def create_tour(self, title):
+        return self.__rpc("create_tour", [title])
+
+    def open_tour(self, path, edit):
+        return self.__rpc("open_tour", [path, edit])
+
+    def set_tour_edit(self, tour_id, edit):
+        return self.__rpc("set_tour_edit", [tour_id, edit])
+
+    def view_tour(self, tour_id):
+        return self.__rpc("view_tour", [tour_id])
+
+    def edit_tour_metadata(self, tour_id, delta):
+        return self.__rpc("edit_tour_metadata", [tour_id, delta])
+
+    def forget_tour(self, tour_id):
+        return self.__rpc("forget_tour", [tour_id])
+
+    def create_stop(self, tour_id, title, path, line):
```

Comment:
Can you add an `index` param? Need it to enable undoing deletions.

Reply:
Index is supposed to be internal state of the server. Why do you need that in order to undo deletions?

Reply:
If I delete the 2nd stop in a 3-stop tour, how do I undo that operation?

Reply:
Should probably just provide a basic undo log for delete operations, but now I also get what you meant. I read index as "repo index" and was confused about what that had to do with anything. I totally forgot to add move and relocate commands for stops, so that's important. Those will have index params lol